recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -48,74 +42,98 @@ def _minimum_compilers_version(self):
             "clang": "6",
         }
 
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder,
+                 self.export_sources_folder)
+
     def configure(self):
+        if Version(self.version) < "0.9.0":
+            self.license = "LGPL-2.1-or-later"
+
         if self.options.shared:
             del self.options.fPIC
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if self.info.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.info.settings.compiler))
         if not min_version:
             self.output.warn("{} recipe lacks information about the {} compiler support.".format(
-                self.name, self.settings.compiler))
+                self.name, self.info.settings.compiler))
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
+            if Version(self.info.settings.compiler.version) < min_version:
                 raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
-        if self.settings.os != "Linux":
+                    self.name, self._minimum_cpp_standard, self.info.settings.compiler, self.info.settings.compiler.version))
+        if self.info.settings.os != "Linux":
             raise ConanInvalidConfiguration("Only Linux supported")
 
     def build_requirements(self):
-        self.build_requires("pkgconf/1.7.4")
+        self.tool_requires("pkgconf/1.7.4")
         if self.options.with_code_gen:
-            self.build_requires("expat/2.4.1")
+            self.tool_requires("expat/2.4.8")
 
     def requirements(self):
-        self.requires("libsystemd/249.4")
+        self.requires("libsystemd/251.4")
+
+    def layout(self):
+        cmake_layout(self)
```

Comment:
```suggestion
        cmake_layout(self, src_folder="src")
```
Prefer separate source from build, so you can analyze when building your package